src/network/core/udp.cpp

```diff
@@ -134,14 +138,14 @@ void NetworkUDPSocketHandler::ReceivePackets()
 #endif
 
 			NetworkAddress address(client_addr, client_len);
-			p.PrepareToRead();
 
 			/* If the size does not match the packet must be corrupted.
 			 * Otherwise it will be marked as corrupted later on. */
-			if (nbytes != p.size) {
+			if (!p.ParsePacketSize() || nbytes != p.Size()) {
```

Comment:
What's the best way to get rid of these warnings? I know a cast will do, but nbytes really needs to be ssize_t because the transfer functions can return negative values and Size() is effectively vector::size() thus size_t.

Reply:
It's reasonable to assume that vector::size won't return larger than will fit into a ssize_t, so it's probably better to cast that and do the comparison as signed.